unix_integration/src/resolver.rs

```diff
@@ -788,18 +823,29 @@ where
     async fn offline_account_authenticate(
         &self,
         token: &Option<UserToken>,
-        cred: &str,
-    ) -> Result<Option<bool>, ()> {
+        cred: Option<PamCred>,
+    ) -> Result<ClientResponse, ()> {
+        let cred = match cred {
+            Some(cred) => match cred {
+                PamCred::Password(v) => v.clone(),
+                // We can only authenticate using a password
+                _ => return Ok(ClientResponse::PamStatus(Some(false))),
```

Comment:
I wasn't certain how to handle this case exactly. If we were in the middle of a authentication attempt, and suddenly went offline, then we could be stuck with an MFA token and no way to perform an offline logon. I've chosen to simply reject the auth attempt, but we could instead return a request for a password prompt here (such as you see on line 835). The only downside is the user could be prompted multiple times for the password. Also, if we come back online while waiting for that additional password, we'll have to start all over again. This could technically go on indefinitely, if there is some network instability.